perun-core/src/main/java/cz/metacentrum/perun/core/impl/ConsentsManagerImpl.java

```diff
@@ -53,11 +63,289 @@
+		if(rs.getInt("consents_created_by_uid") == 0) c.setCreatedByUid(null);
+		else c.setCreatedByUid(rs.getInt("consents_created_by_uid"));
+		return c;
+	};
+
+	private final static RowMapper<Consent> CONSENT_MAPPER_WITHOUT_HUB = (rs, rowNum) -> {
+		Consent c = new Consent();
+		c.setId(rs.getInt("consents_id"));
+		c.setUserId(rs.getInt("consents_user_id"));
+		c.setConsentHub(new ConsentHub());
+		c.setStatus(ConsentStatus.valueOf(rs.getString("consents_status")));
+		c.setCreatedAt(rs.getString("consents_created_at"));
+		c.setCreatedBy(rs.getString("consents_created_by"));
+		c.setModifiedAt(rs.getString("consents_modified_at"));
+		c.setModifiedBy(rs.getString("consents_modified_by"));
+		if(rs.getInt("consents_modified_by_uid") == 0) c.setModifiedByUid(null);
+		else c.setModifiedByUid(rs.getInt("consents_modified_by_uid"));
+		if(rs.getInt("consents_created_by_uid") == 0) c.setCreatedByUid(null);
+		else c.setCreatedByUid(rs.getInt("consents_created_by_uid"));
+		return c;
+	};
+
+
+
+
 	public ConsentsManagerImpl(DataSource perunPool) {
 		jdbc = new JdbcPerunTemplate(perunPool);
 		jdbc.setQueryTimeout(BeansUtils.getCoreConfig().getQueryTimeout());
 	}
 
+	@Override
+	public Consent createConsent(PerunSession perunSession, Consent consent) throws ConsentNotExistsException, ConsentExistsException, ConsentHubNotExistsException {
+		// Check if consent already exists
+		if(consentExists(perunSession, consent)){
+			throw new ConsentExistsException("Consent already exists.");
+		}
+
+		try {
+			int newId = Utils.getNewId(jdbc, "consents_id_seq");
+
+			jdbc.update("insert into consents(id, user_id, consent_hub_id, status, created_at, created_by, modified_at, modified_by, created_by_uid, modified_by_uid) " +
+					"values (?,?,?,?::consent_status,"+Compatibility.getSysdate()+",?,"+Compatibility.getSysdate()+",?,?,?)", newId, consent.getUserId(),
+				consent.getConsentHub().getId(), ConsentStatus.UNSIGNED.toString(), perunSession.getPerunPrincipal().getActor(), perunSession.getPerunPrincipal().getActor(),
+				perunSession.getPerunPrincipal().getUserId(), perunSession.getPerunPrincipal().getUserId());
+			log.info("Consent {} created.", consent);
+			consent.setId(newId);
+			createAttrDefsForConsent(perunSession, consent);
+			return consent;
+		} catch (Exception e){
+			throw new InternalErrorException(e);
+		}
+	}
+
+	@Override
+	public void deleteConsent(PerunSession perunSession, Consent consent) throws ConsentNotExistsException {
+		try {
+			if (!consentExists(perunSession, consent)) {
+				throw new ConsentNotExistsException("Consent: " + consent + " doesn't exists.");
+			}
+			removeAttrsForConsent(perunSession, consent.getId());
+			consent.setAttributes(null);
+
+			int numberOfRows = jdbc.update("delete from consents where id=?", consent.getId());
+			if (numberOfRows != 1) throw new ConsentNotExistsException(consent);
+			log.info("Consent {} deleted.", consent);
+		} catch (Exception e){
```

Comment:
i think we don't want to return any thrown exception in the block as ConsentNotExistsException

Reply:
We should also catch here only runtime exceptions and rethrow them as `InternalErrorException`. Otherwise, `ConsentNotExistsException` from two lines above would also be caught here and rethrown again.